src/legacy/core_plugins/table_vis/public/plugin.ts

```diff
@@ -41,8 +41,7 @@ export interface TableVisualizationDependencies extends LegacyDependenciesPlugin
 
 /** @internal */
 export interface TablePluginSetupDependencies {
-  // TODO: Remove `any` as functionsRegistry will be added to the DataSetup.
-  data: DataSetup | any;
+  data: ReturnType<DataPublicPlugin['setup']>;
```

Comment:
👍  As discussed we'll update this later when we sort out the discrepancy between the src/plugins/data and src/legacy/core_plugins/data interfaces